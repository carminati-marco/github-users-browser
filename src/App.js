import React from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import styled from "styled-components";

import { Container, Row, Col } from "react-bootstrap";
import logo from "./logo.png";
import UserList from "./components/UserList";
import UserDetail from "./components/UserDetail/index";

const Header = styled.header`
  background-color: #282c34;
  min-height: 72px;
  line-height: 72px;

  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: left;
  color: #fff;

  &&&& img {
    padding: 4px;
  }
`;

const StyledContainer = styled(Container)`
  padding: 10px;
`;

function App() {
  return (
    <div>
      <Header>
        <img src={logo} alt="Github" />
      </Header>
      <StyledContainer fluid>
        <Row>
          <Col xs lg="3">
            <UserList />
          </Col>
          <Col>
            <UserDetail />
          </Col>
        </Row>
      </StyledContainer>
    </div>
  );
}

export default App;
