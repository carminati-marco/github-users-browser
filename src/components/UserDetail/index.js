import React from "react";
import styled from "styled-components";
import { Card, ListGroup } from "react-bootstrap";

const StyledCardImg = styled(Card.Img)`
  width: 180px;
  height: 180px;
  border: 1px solid #000;
`;

const StyledCardBody = styled(Card.Body)`
  padding: 10px;
  display: flex;
  align-items: flex-start;

  div {
    margin-left: 5px;
  }
`;

const UserDetail = () => (
  <Card>
    <StyledCardBody>
      <StyledCardImg variant="top" src="https://github.githubassets.com/images/modules/logos_page/Octocat.png" />
      <Card.Title>
        Marco<br />Carminati
      </Card.Title>
    </StyledCardBody>
    <Card.Body>
      <Card.Title>Repositories</Card.Title>
      <ListGroup>
        <ListGroup.Item>Linus Torvalds</ListGroup.Item>
        <ListGroup.Item>Marco Carminati</ListGroup.Item>
        <ListGroup.Item>Another developer</ListGroup.Item>
      </ListGroup>
    </Card.Body>
  </Card>
);
export default UserDetail;
