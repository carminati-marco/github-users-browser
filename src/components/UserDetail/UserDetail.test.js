import React from "react";
import { Provider } from "react-redux";

import store from "../../redux/store";
import UserDetail from "./index";

import { mount } from "enzyme";
import App from "../../App";

describe("UserDetail test", () => {
  let component;

  beforeEach(() => {
    component = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it("renders correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("Cart must be 0", () => {
    expect(component.find(UserDetail).find("Card").length).toEqual(0);
  });

  it("ListGroupItem toggle activate", () => {
    component
      .find("ListGroupItem")
      .first()
      .simulate("click");

    component.update();
    expect(component.find(UserDetail).find("Card").length).toEqual(1);
    expect(component).toMatchSnapshot();

    component
      .find("ListGroupItem")
      .first()
      .simulate("click");

    component.update();
    expect(component.find(UserDetail).find("Card").length).toEqual(0);
  });
});
