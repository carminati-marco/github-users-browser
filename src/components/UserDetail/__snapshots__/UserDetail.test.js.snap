// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail test ListGroupItem toggle activate 1`] = `ReactWrapper {}`;

exports[`UserDetail test renders correctly 1`] = `ReactWrapper {}`;
